### PR TITLE
[GOBBLIN-1221] Preserve source file's ModTime by configuration

### PR DESCRIPTION
**Pull request by autumnust**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [ ] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1221

### Description
This change enabled distcp to preserve the ModTime in destination. It refactored a code block in `CopyPublisher` which is used to preserve the file version. Note that this is required to be placed in publisher since the rename in the publisher is the last operation change the ModTime, while other attributes like Owner/Group will not change after write operation in writer. 

Add a unit test for both positive and negative case.

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/publisher/CopyDataPublisher.java [hunk 1]
```diff
@@ -185,9 +185,39 @@ public void handleUnpublishedWorkUnits(Collection<? extends WorkUnitState> state
...
+  /**
+   * Unlike other preserving attributes of files (ownership, group, etc.), which is preserved in writer,
+   * some of the attributes have to be set during publish phase like ModTime,
+   * and versionStrategy (usually relevant to modTime as well), since they are subject to change with Publish(rename)
+   */
+  private void preserveFileAttrInPublisher(CopyableFile copyableFile) throws IOException {
+    // Preserving File ModTime, keep the accessing time of the target file.
+    if (copyableFile.getPreserve().preserve(PreserveAttributes.Option.MOD_TIME)) {
+      fs.setTimes(copyableFile.getDestination(), copyableFile.getOriginTimestamp(),
+          fs.getFileStatus(copyableFile.getDestination()).getAccessTime());
```
Comment (sv2000) [inline gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/publisher/CopyDataPublisher.java:197]: Why not add accessTime as a field in CopyableFile and populate it when CopyableFiles are computed? We can avoid the fs.getFileStatus() RPC call here. Also, not sure whether we should preserve the file "accessTime" from the source. We might be better off setting this to -1.

Comment (autumnust) [reply gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/publisher/CopyDataPublisher.java]: Actually I think a bit more into the accessTime, it might makes more sense to set this to -1 since the preservation of ModTime is actually breaking the semantics already (we preserve it for special use cases only) and it doesn't make sense to keep the original access time either.
